fhd_core/source_modeling/vis_model_transfer.pro

```diff
@@ -11,6 +14,76 @@ function vis_model_transfer,obs,model_transfer
     print, "Model visibilities transferred from " + transfer_name
   endfor
 
+  ;; Get the params file associated with the model visibilities
+  if file_test(model_transfer+'/'+obs.obsname+'_params.sav') then begin
+    params_model = getvar_savefile(model_transfer+'/'+obs.obsname+'_params.sav','params')
+  endif else begin
+    if file_test(file_dirname(model_transfer)+'/metadata/'+obs.obsname+'_params.sav') then begin
+      params_model = getvar_savefile(file_dirname(model_transfer)+'/metadata/'+obs.obsname+'_params.sav','params')
+    endif else message, 'No params file found in model transfer directory.'
+  endelse 
+  model_n_time = n_elements(uniq(params_model.time))
+  model_nbaselines = (uniq(params_model.time))[0]+1
+
+  if model_n_time NE data_n_time then begin
+  ;; Exclude flagged times from the model visibilities if the number of time steps do not match
+
+    data_Jdate = (*obs.baseline_info).Jdate
+    time_res_Jdate = (double(obs.time_res)/2.) / (24.*3600.) ;offset to the center of a time step
+    model_time = params_model.time[uniq(params_model.time)]
+    data_time = params.time[uniq(params.time)]
+    tolerance = 1E-5
+
+    ; Match times betweeen model and data
+    ; Option 1: You are working with a model made by FHD. Thus the timing convention in the params is the same
+    ; Option 2: You are working with a model made by WODEN. Thus the timing convention in the params is different
+    ;           and you need to convert the model Jdate to the data Jdate within precision.
+
+    ; Option 1
+    matched_times_opt1 = intarr(data_n_time)
+    for data_time_i=0, data_n_time-1 do begin
+      min_val= min(abs(model_time - data_time[data_time_i]), min_ind)
+      if min_val LT tolerance then matched_times_opt1[data_time_i] = min_ind
+    endfor
+
+    ; Option 2
+    matched_times_opt2 = intarr(data_n_time)
+    for data_time_i=0, data_n_time-1 do begin
+      min_val = min(abs(model_time - (data_Jdate[data_time_i]-time_res_Jdate)), min_ind)
+      if min_val LT tolerance then matched_times_opt2[data_time_i] = min_ind
+    endfor
+
+    ; Find which option was more successful in matching the times
+    if total(matched_times_opt1 < 1) GT total(matched_times_opt2 < 1) then matched_times = matched_times_opt1
+    if total(matched_times_opt1 < 1) LT total(matched_times_opt2 < 1) then matched_times = matched_times_opt2
```

Comment:
what if neither are very good? should we error or warn if the best one is still way off?

Reply:
I've included two errors to be explicit, one if no times matched and one if the times matched do not equal the number of timesteps in the data.